spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/KafkaBackoffAwareMessageListenerAdapter.java

```diff
@@ -94,7 +96,12 @@ public void onMessage(ConsumerRecord<K, V> consumerRecord, @Nullable Acknowledgm
 				.ifPresent(nextExecutionTimestamp -> this.kafkaConsumerBackoffManager
 						.backOffIfNecessary(createContext(consumerRecord, nextExecutionTimestamp, consumer)));
 		try {
-			invokeDelegateOnMessage(consumerRecord, acknowledgment, consumer);
+			Acknowledgment ack = acknowledgment;
+			if (ack == null) {
+				// The default DLT handler now requires an Acknowledgment.
```

Comment:
but I see it is `@Nullable` there. So, what is wrong to continue to use `acknowledgment` as it is?

Reply:
Because of this

https://github.com/spring-projects/spring-kafka/blob/8efd15fe8e4791ab69d0dfc3893cfb4902058868/spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/MessagingMessageListenerAdapter.java#L370-L372

and this

https://github.com/spring-projects/spring-kafka/blob/8efd15fe8e4791ab69d0dfc3893cfb4902058868/spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/MessagingMessageListenerAdapter.java#L386-L392

I changed the reporter's reproducer to use default `AckMode` and hit this problem.

I didn't think it was worth it to remove the null check in the method, but I can do so if you insist,

Reply:
Hmmm - maybe I could see if the parameter is annotated `@Nullable` and skip that check - will look at it tomorrow.

Reply:
No; that won't work; the invoke already failed.

Reply:
I guess I would have to re-invoke the method in that case, since the invoke already failed.

Reply:
Correction: I can check the nullability before the first invocation.